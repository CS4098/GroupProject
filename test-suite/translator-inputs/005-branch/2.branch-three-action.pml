process test{
	branch br1 {
		action act_1 {
		requires { a }
		provides { b }
		}
		action act_2 {
		requires { b }
		provides { c }
		}
		action act_3 {
		requires { a }
		provides { c }
		}
	}
}
