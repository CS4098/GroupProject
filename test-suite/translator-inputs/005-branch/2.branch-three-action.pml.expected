bool a;
bool b;
bool c;

active proctype test()
{
	int br1 = _nr_pr;
	run act_1();
	run act_2();
	run act_3();
	_nr_pr == br1 ->
}

proctype act_1()
{
	a ->
		b = true;
}

proctype act_2()
{
	b ->
		c = true;
}

proctype act_3()
{
	a ->
		c = true;
}
